ros/processor/suggestions_engine.py

```diff
@@ -64,25 +93,125 @@ def run(self):
             self.consumer.close()
 
 
-def download_and_extract(service, event, archive_URL, host, org_id):
-    logging.info(f"{service} - {event} - Downloading the report for system {host.get('id')}.")
+def find_root_directory(directory, target_file):
+    """
+    Recursively search for the target file in the given directory - to get root directory of an archive.
+    """
+    for dirpath, dirnames, filenames in os.walk(directory):
+        if target_file in filenames:
+            return dirpath
+
+    return None
+
+
+def get_index_file_path(service, event, host, extracted_dir):
+    extracted_dir_root = find_root_directory(extracted_dir, "insights_archive.txt")
+
+    if not extracted_dir_root:
+        logging.error(
+            f"{service} - {event} - insights_archive.txt not found in the extracted dir for system {host.get('id')}."
+        )
+        return None
+
+    pmlogger_dir = os.path.join(extracted_dir_root, "data/var/log/pcp/pmlogger/")
+
+    index_files = [
+        file for file in os.listdir(pmlogger_dir) if file.endswith(".index")
+    ]
+    index_file_path = os.path.join(pmlogger_dir, index_files[0])
+
+    return index_file_path
+
 
-    response = requests.get(archive_URL, timeout=10)
+def create_output_dir(request_id, host):
+    output_dir = f"/tmp/pmlogextract-output-{request_id}/"
```

Comment:
Though we need the directory and files for temporary purpose however these files are getting passed to next command `pmlogsummary`, so in this situation I don't think using `/tmp` is wise idea. The initial impression says the `/tmp` is managed by `systemd-tmpfiles-clean.service` so its possible that these could be deleted before `pmlogsummary` requires it.
Should we look into adding small persistent volume? maybe 250Mb is enough? and there we do these file operations?

Reply:
In that case, we can use `/var/tmp` .

Reply:
I need to check! If I understand correctly the insights-core does not have duties about collecting data from containers? those are instances of VM?

Reply:
The `/var/tmp` is better choice, I do see security concerns as well when using these directories,
https://www.redhat.com/en/blog/polyinstantiating-tmp-and-vartmp-directories

I have captured both the usage and the security concern in new Jira ticket RHINENG-15674 and we can evaluate it later. As of now lets go with `/var/tmp`.